src/link/SpirV/deduplicate.zig

```diff
@@ -217,7 +218,7 @@ const EntityContext = struct {
         // If the current pointer is recursive, don't immediately add it to the map. This is to ensure that
         // if the current pointer is already recursive, it gets the same hash a pointer that points to the
         // same child but has a different result-id.
-        if (entity.kind == .OpTypePointer) {
```

Comment:
Again, these should be emitted for Vulkan too. Physical Storage Buffer pointers are for example allowed to have recursive types. If the frontend emits a recursive type thats a frontend error, and we shouldn't silently drop it here and create an invalid module.